services/meta/client.go

```diff
@@ -639,6 +639,21 @@ func (c *Client) ShardsByTimeRange(sources influxql.Sources, tmin, tmax time.Tim
 	return a, nil
 }
 
+// DropShard deletes a shard by ID.
+func (c *Client) DropShard(id uint64) error {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	data := c.cacheData.Clone()
+
+	if err := data.DropShard(id); err != nil {
+		return err
+	}
+
+	c.commit(data)
```

Comment:
Seems to be ignored on every other call in the meta client too. Should they all be updated to check? @corylanou?